internal/test/wsu/wsu_test.go

```diff
@@ -242,13 +407,10 @@ func testNoPendingCSRs(t *testing.T) {
 }
 
 // testWorkerLabelsArePresent tests if the worker labels are present on the Windows Node.
-func testWorkerLabelsArePresent(t *testing.T) {
-	// Check if the Windows node has the required label needed.
-	windowsNodes, err := framework.K8sclientset.CoreV1().Nodes().List(metav1.ListOptions{LabelSelector: windowsLabel})
-	require.NoErrorf(t, err, "error while getting Windows node: %v", err)
-	assert.Equalf(t, len(windowsNodes.Items), vmCount, "expected %d Windows node(s) to be present but found %v",
-		vmCount, len(windowsNodes.Items))
-	assert.Contains(t, windowsNodes.Items[0].Labels, workerLabel,
+func testWorkerLabelsArePresent(t *testing.T, vm e2ef.WindowsVM) {
+	node, err := getNode(vm.GetCredentials().GetIPAddress())
```

Comment:
You are making this call in multiple places. Rather, make this call once and pass the `node` object into the functions.